django/db/migrations/state.py

```diff
@@ -548,7 +549,7 @@ def render(self, apps):
         body.update(self.construct_managers())
 
         # Then, make a Model object (apps.register_model is called in __new__)
-        return type(
+        return metaclassmaker()(
```

Comment:
Since `type()` is responsible for more than 50% of the current time consumption (https://code.djangoproject.com/ticket/24743#comment:15) I'm -1 on this change right now, knowing that there is a bug that cannot otherwise be fixed (from what I see). But I don't think we can accept any more increased in time here, as this would affect the majority of users while custom meta classes are likely used by only very view people. Sorry!

I appreciate your work on this, though. If we find away to make migrations faster I'm happy to reconsider this patch and would like you to reopen a pull request then. Thank you!